File: LC/rough/local_practice/p5.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int l = 0;
        int r = nums.size() - 1;
        int mid;
        while (l <= r)
        {
            mid = l + (r - l) / 2;

            if (nums[mid] == target)
            {
                return mid;
            }
            else if (target > nums[mid])
            {
                l = mid + 1;
            }
            else
            {
                r = mid - 1;
            }
        }

        return -1;
    }
};

File: LC/rough/local_practice/p5.py
class Solution:
    def minSubArrayLen(self, target: int, nums: list[int]) -> int:
        size = len(nums)
        left = 0
        sum = 0
        minSub = float("inf")

        for right in range(size):
            sum += nums[right]

            while sum >= target:
                minSub = min(minSub, right - left + 1)
                sum -= nums[left]
                left += 1

        return minSub if minSub != float("inf") else 0
